code/controllers/subsystem/daynightcycle.dm

```diff
@@ -0,0 +1,111 @@
+var/datum/subsystem/daynightcycle/SSDayNight
+var/list/daynight_turfs = list()
+/* Original Plan
+Morning	  - 2 Mins
+Sunrise   - 2 Mins
+Daytime   - 16 Minutes
+Afternoon - 16 Minutes
+Sunset    - 2 Minutes
+Nighttime - 36 Minutes
+*/
+					
+#define TOD_MORNING 	"#4d6f86" 
+#define TOD_SUNRISE 	"#fdc5a0"
+#define TOD_DAYTIME 	"#FFFFFF"
+#define TOD_AFTERNOON 	"#ffeedf"
+#define TOD_SUNSET 		"#75497e"
+#define TOD_NIGHTTIME 	"#001522"
+
+/datum/subsystem/daynightcycle
+	name          = "Day Night Cycle"
+	init_order    = SS_INIT_DAYNIGHT
+	display_order = SS_DISPLAY_DAYNIGHT
+	priority      = SS_PRIORITY_DAYNIGHT
+	wait          = 1 MINUTES
+	flags         = SS_FIRE_IN_LOBBY
+
+	var/current_timeOfDay = TOD_DAYTIME //We start tickers maxed out, and start on afternoon
+	var/next_light_power = 10
+	var/next_light_range = 1
+
+	//The initial values don't matter, it just needs to fire initially, then set itself into the cycle.
+	var/next_firetime = 0
+	var/list/currentrun
+
+/datum/subsystem/daynightcycle/New()
+	NEW_SS_GLOBAL(SSDayNight)
+
+/datum/subsystem/daynightcycle/Initialize()
+	get_turflist()
+	..()
+
+/datum/subsystem/daynightcycle/fire(resumed = FALSE)
+	if(flags & SS_NO_FIRE)
+		return
+	if(!map.daynight_cycle)
+		flags |= SS_NO_FIRE
+		pause()
+	else
+		if(world.time >= next_firetime)
+			switch(current_timeOfDay) //Then set the next segment up.
+				if(TOD_MORNING)
+					current_timeOfDay = TOD_SUNRISE
+					next_firetime = world.time + 3 MINUTES
+					play_globalsound()
+				if(TOD_SUNRISE)
+					current_timeOfDay = TOD_DAYTIME
+					next_firetime = world.time + 14 MINUTES
+				if(TOD_DAYTIME)
+					current_timeOfDay = TOD_AFTERNOON
+					next_firetime = world.time + 15 MINUTES
+				if(TOD_AFTERNOON)
+					current_timeOfDay = TOD_SUNSET
+					next_firetime = world.time + 3 MINUTES
+				if(TOD_SUNSET)
+					current_timeOfDay = TOD_NIGHTTIME
+					next_firetime = world.time + 36 MINUTES
+					play_globalsound()
+				if(TOD_NIGHTTIME)
+					current_timeOfDay = TOD_MORNING
+					next_firetime = world.time + 5 MINUTES
+				
+
+			currentrun = daynight_turfs.Copy()
```

Comment:
It should only do this if `resumed` is false. Otherwise, looks fine to me.

Reply:
I guess that's kind of implied by the clause this is in, but it would be nice for readability to change it anyway

Reply:
Yeah I made the change